tools/convert_datasets/mot2reid.py

```diff
@@ -0,0 +1,169 @@
+            gt = gt.strip().split(',')
+            frame_id, ins_id = map(int, gt[:2])
+            ltwh = list(map(float, gt[2:6]))
+            class_id = int(gt[7])
+            visibility = float(gt[8])
+            if class_id in USELESS:
+                continue
+            elif class_id in IGNORES:
+                continue
+            elif visibility < args.vis_threshold:
+                continue
+            reid_img_folder = osp.join(reid_train_folder,
+                                       f'{video_name}_{ins_id:06d}')
+            if not osp.exists(reid_img_folder):
+                os.makedirs(reid_img_folder)
+            idx = len(os.listdir(reid_img_folder))
+            reid_img_name = f'{idx:06d}.jpg'
+            if frame_id != last_frame_id:
+                raw_img_name = raw_img_names[frame_id - 1]
+                raw_img = mmcv.imread(
+                    f'{video_folder}/{raw_img_folder}/{raw_img_name}')
+                last_frame_id = frame_id
+            xyxy = np.asarray(
+                [ltwh[0], ltwh[1], ltwh[0] + ltwh[2], ltwh[1] + ltwh[3]])
+            reid_img = mmcv.imcrop(raw_img, xyxy)
+            mmcv.imwrite(reid_img, f'{reid_img_folder}/{reid_img_name}')
+
+    reid_meta_folder = osp.join(args.output, 'meta')
+    if not osp.exists(reid_meta_folder):
+        os.makedirs(reid_meta_folder)
+    reid_train_list = []
+    reid_val_list = []
+    reid_img_folder_names = os.listdir(reid_train_folder)
+    sorted(reid_img_folder_names)
+    ids_num = len(reid_img_folder_names)
+    train_ids_num = int(ids_num * (1 - args.val_split))
+    train_label, val_label = 0, 0
+    random.seed(0)
+    for reid_img_folder_name in reid_img_folder_names[:train_ids_num]:
+        reid_img_names = os.listdir(
+            f'{reid_train_folder}/{reid_img_folder_name}')
+        # ignore ids whose number of image is less than min_per_person
+        if (len(reid_img_names) < args.min_per_person):
+            continue
+        # downsampling when there are too many images owned by one id
+        if (len(reid_img_names) > args.max_per_person):
+            reid_img_names = random.sample(reid_img_names, args.max_per_person)
+        sorted(reid_img_names)
+        for reid_img_name in reid_img_names:
+            reid_train_list.append(
+                f'{reid_img_folder_name}/{reid_img_name} {train_label}\n')
+        train_label += 1
+    for reid_img_folder_name in reid_img_folder_names[train_ids_num:]:
+        reid_img_names = os.listdir(
+            f'{reid_train_folder}/{reid_img_folder_name}')
+        sorted(reid_img_names)
+        for reid_img_name in reid_img_names:
+            reid_val_list.append(
+                f'{reid_img_folder_name}/{reid_img_name} {val_label}\n')
+        val_label += 1
+    with open(osp.join(reid_meta_folder, 'train.txt'), 'w') as f:
+        f.writelines(reid_train_list)
+    with open(osp.join(reid_meta_folder, 'val.txt'), 'w') as f:
+        f.writelines(reid_val_list)
```

Comment:
Think about how to run the script once, and you can get all metafiles you need.
For example, you can split the file to train and val, so that for validation, but you still need a train meta file for training.

Reply:
Add reid_research_list to the save whole dataset, which saved in `osp.join(reid_meta_folder, 'research.txt')`. In this case, all metafiles can be obtained in one loop.